### PR TITLE
Highlights attribute matches

### PR DESCRIPTION
**Pull request by rg-coder**
Highlights the serialized attribute matches with @exist:matches
attribute in KWIC summarized response.

### CODE REVIEW

**Comment by rg-coder** [comment]
After the Adam's pull request #https://github.com/eXist-db/exist/pull/1130 ,if suppose the TEI document matches the lemma attribute "commit" and gives the desired result after expanding the serialized result as follows:

```xml
<w xmlns="http://www.tei-c.org/ns/1.0" xmlns:exist="http://exist.sourceforge.net/NS/exist" lemma="commit" exist:matches="lemma">committed</w>
```

Then after, passing it through KWIC summary, user will have the output as follows:

```xml
<p>
<span class="previous">... KWIC highlighting code has been successfully </span>
<span class="hi">committed</span>
<span class="following"> to the github repository ...</span>
</p>

**Comment by joewiz** [comment]
@rg-coder Cool! I'm curious, can you give an example of a query that would yield the match on your `<tei:w>` element?

**Comment by rg-coder** [comment]
@joewiz Please find the example below:

### test.xml
```xml
<?xml version="1.0"?>
<TEI xmlns="http://www.tei-c.org/ns/1.0">
	<p>
		<s>
			<w>KWIC</w><w>highlighting</w>
			<w>code</w><w>has</w>
			<w>been</w>
			<w lemma="success">successfully</w>
			<w lemma="commit">committed</w>
			<w>to</w><w>the</w><w>github</w><w>repository</w>!
		</s>
	</p>
</TEI>
```

### collection.xconf
```xml
<collection xmlns="http://exist-db.org/collection-config/1.0">
    <index xmlns:tei="http://www.tei-c.org/ns/1.0" xmlns:xs="http://www.w3.org/2001/XMLSchema">
        <lucene>
            <text qname="@lemma" type="xs:string"/>
        </lucene>
    </index>
</collection>
```

### test.xq
```xml
for $hit in collection("/db/test")//tei:p[.//tei:w[ft:query(@lemma, <query><term>commit</term></query>)]]
let $expanded :=  util:expand($hit, "highlight-matches=both")
return kwic:get-summary($expanded, $expanded//*[@exist:matches], <config width="120"/>)
```

### test-results
```xml
<p>
<span class="previous"> KWIC highlighting code has been successfully </span>
<span class="hi">committed</span>
<span class="following"> to the github repository ! </span>
</p>
```

**Comment by joewiz** [comment]
@rg-coder Thank you! Very helpful.

File: src/org/exist/xquery/lib/kwic.xql [hunk 1]
```diff
@@ -189,25 +189,26 @@ declare function kwic:get-summary($root as node(), $node as element(exist:match)
...
     Expand the element in $hit. Creates an in-memory copy of the element and marks
-    all matches with an exist:match tag, which will be used by all other functions in
+    all element matches with an exist:match tag and attribute matches with an 
+    exist:matches tag, which will be used by all other functions in
     this module. You need to call kwic:expand before kwic:get-summary. 
     kwic:summarize will call it automatically.
 :)
 declare function kwic:expand($hit as element()) as element() {
-    util:expand($hit, "expand-xincludes=no")
+    util:expand($hit, "highlight-matches=both")
```
Comment (adamretter) [inline src/org/exist/xquery/lib/kwic.xql]: Can you change this to: `"expand-xincludes=no highlight-matches=both"`

File: src/org/exist/xquery/lib/kwic.xql [hunk 2]
```diff
@@ -240,8 +241,8 @@ declare function kwic:summarize($hit as element(), $config as element(config)?)
 :)
 declare function kwic:summarize($hit as element(), $config as element(config)?, 
     $callback as (function(node(), xs:string) as xs:string?)?) as element()* {
-    let $expanded := util:expand($hit, "expand-xincludes=no")
-	for $match in $expanded//exist:match
+    let $expanded := util:expand($hit, "highlight-matches=both")
```
Comment (adamretter) [inline src/org/exist/xquery/lib/kwic.xql]: Also this one should be `"expand-xincludes=no highlight-matches=both"`

Comment (rg-coder) [inline src/org/exist/xquery/lib/kwic.xql]: Thanks Adam for the review comments. I have fixed it now.

**Comment by adamretter** [comment]
Thank you very much for your contribution @rg-coder

**Pull request merged**